walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -265,10 +355,52 @@ public void notEnoughRoom() {
 
     MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_60_MINUTES);
 
-    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> actual = query.query(events, request, NO_ATTENDEES);
     Collection<TimeRange> expected = Arrays.asList();
 
     Assert.assertEquals(expected, actual);
   }
-}
 
+  @Test
+  public void noMandatoryAttendees() {
+    // No mandatory attendees, just two optional attendees with one gap in their schedules.
```

Comment:
Good - it is always sensible to test the "boundary cases" of our business logic.

So classic boundary cases are things like -1,0,1, MAX_INTEGER etc - i.e. you are testing  all the places where the logic might fall apart because you are testing outside the limits of what you expect to get, rather than just the "happy day" scenario where you just test with "normal" data.

Perhaps a couple of other boundary conditions that I am not sure are covered already could be:
 - what if there are no events?
 - what if the day is only a half-day? or zero length? (e.g. weekend day?)
 - what if the events last longer than the day?

Sometimes it feels like a waste of time to test these boundary cases because "this can never happen" ... but of course we know that with code sometimes things we think cannot happen actually do :-)  Especially in the future when we come back to make a change or fix a bug and we've forgotten what we were doing - those boundary checks in our tests might save us from making a big mistake.

Reply:
Thank you for the tip. I added a test with no events and two that have events lasting more than the day. 

I'm not sure if the case when the day is only a half-day or zero length would be valid, but I will think about it.